modules/core/04-channel/types/timeout_test.go

```diff
@@ -0,0 +1,131 @@
+package types_test
+
+import (
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	"github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+)
+
+func (suite *TypesTestSuite) TestTimeoutPassed() {
+	var (
+		timeout types.Timeout
+		passed  bool
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"client height is not after timeout height",
+			func() {
+				passed = timeout.AfterHeight(clienttypes.NewHeight(0, 75))
+			},
+			false,
+		},
+		{
+			"client timestamp is not after timeout timestamp",
+			func() {
+				passed = timeout.AfterTimestamp(75)
+			},
+			false,
+		},
+		{
+			"client height is after timeout height",
+			func() {
+				passed = timeout.AfterHeight(clienttypes.NewHeight(0, 25))
+			},
+			true,
+		},
+		{
+			"client timestamp is after timeout timestamp",
+			func() {
+				passed = timeout.AfterTimestamp(25)
+			},
+			true,
+		},
+		{
+			"client timestamp equal to timeout timestamp",
+			func() {
+				passed = timeout.AfterTimestamp(50)
+			},
+			false,
+		},
+		{
+			"client block height equal to timeout block height",
+			func() {
+				passed = timeout.AfterHeight(clienttypes.NewHeight(0, 50))
+			},
+			false,
+		},
+		{
+			"client block height equal to timeout block height, with different revision height",
+			func() {
+				passed = timeout.AfterHeight(clienttypes.NewHeight(1, 50))
+			},
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			timeout = types.NewTimeout(
+				clienttypes.NewHeight(0, 50),
+				50,
+			)
+
+			tc.malleate()
+
+			if tc.expPass {
+				suite.Require().True(passed)
+			} else {
+				suite.Require().False(passed)
+			}
+		})
+	}
+}
+
+func (suite *TypesTestSuite) TestTimeout() {
+	var timeout types.Timeout
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"valid timeout",
+			func() {},
+			true,
+		},
+		{
+			"invalid timeout",
+			func() {
+				timeout.Height = clienttypes.ZeroHeight()
+				timeout.Timestamp = 0
+			},
+			false,
+		},
```

Comment:
we should add some more cases - e.g. height is valid but timestamp is not, as well as the inverse